lib/ansible/modules/cloud/azure/azure_rm_webapp.py

```diff
@@ -0,0 +1,971 @@
+        "id": "/subscriptions/<subscription_id>/resourceGroups/ansiblewebapp1/providers/Microsoft.Web/sites/ansiblewindowsaaa",
+        "kind": "app",
+        "last_modified_time_utc": "2018-05-14T04:50:54.473333Z",
+        "location": "East US",
+        "name": "ansiblewindowsaaa",
+        "outbound_ip_addresses": "52.170.7.25,52.168.75.147,52.179.5.98,52.179.1.81,52.179.4.232",
+        "repository_site_name": "ansiblewindowsaaa",
+        "reserved": false,
+        "resource_group": "ansiblewebapp1",
+        "scm_site_also_stopped": false,
+        "server_farm_id": "/subscriptions/<subscription_id>/resourceGroups/test/providers/Microsoft.Web/serverfarms/plan1",
+        "state": "Running",
+        "tags": {},
+        "type": "Microsoft.Web/sites",
+        "usage_state": "Normal"
+    }
+'''
+
+import time
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from msrestazure.azure_operation import AzureOperationPoller
+    from msrest.serialization import Model
+    from azure.mgmt.web.models import (
+        site_config, app_service_plan, Site,
+        AppServicePlan, SkuDescription, NameValuePair
+    )
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+container_settings_spec = dict(
+    name=dict(type='str', required=True),
+    registry_server_url=dict(type='str'),
+    registry_server_user=dict(type='str'),
+    registry_server_password=dict(type='str')
+)
+
+deployment_source_spec = dict(
+    url=dict(type='str'),
+    branch=dict(type='str')
+)
+
+
+def _normalize_sku(sku):
+    if sku is None:
+        return sku
+
+    sku = sku.upper()
+    if sku == 'FREE':
+        return 'F1'
+    elif sku == 'SHARED':
+        return 'D1'
+    return sku
+
+
+def get_sku_name(tier):
+    tier = tier.upper()
+    if tier == 'F1' or tier == "FREE":
+        return 'FREE'
+    elif tier == 'D1' or tier == "SHARED":
+        return 'SHARED'
+    elif tier in ['B1', 'B2', 'B3', 'BASIC']:
+        return 'BASIC'
+    elif tier in ['S1', 'S2', 'S3']:
+        return 'STANDARD'
+    elif tier in ['P1', 'P2', 'P3']:
+        return 'PREMIUM'
+    elif tier in ['P1V2', 'P2V2', 'P3V2']:
+        return 'PREMIUMV2'
+    else:
+        return None
+
+
+class Actions:
+    NoAction, CreateOrUpdate, UpdateAppSettings, Delete = range(4)
+
+
+class AzureRMWebApps(AzureRMModuleBase):
+    """Configuration class for an Azure RM Web App resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            location=dict(
+                type='str'
+            ),
+            plan=dict(
+                type='raw'
+            ),
+            frameworks=dict(
+                type='list'
```

Comment:
You should be able to define the list value type (dict) and the subspec for the dicts as well

Reply:
resolved